rfd/0201-native-auto-approval.md

```diff
@@ -0,0 +1,364 @@
+---
+authors: Bernard Kim (bernard@goteleport.com)
+state: draft
+---
+
+# RFD 0201 - Native Auto Approval
+
+## Required Approvers
+* Engineering: @r0mant && @fheinecke
+
+## What
+This document describes how Teleport will support native automatic approvals for
+access requests.
+
+## Why
+Currently, Teleport supports automatic approvals of access requests, but support
+is limited. Automatic approvals requires a separate plugin, and only a subset of
+the plugins support automatic approvals. Users requesting this feature do not
+wish to maintain a separate piece of software, or wish to integrate with a
+separate service to utilize this feature. In order to support a wider range of
+use cases, Teleport should support automatic approvals natively.
+
+Automatic approvals enables teams to enforce zero standing privileges, while
+allowing users to get access to their pre approved resources for a limited
+period of time. There are alternative ways to enforce zero standing privileges,
+but a big part of this feature request is to provide a frictionless user
+experience. For example, access lists can be used to achieve similar behavior,
+but a limitation with access lists is that the user must logout and login again
+to renew their certificates to obtain the new roles.
```

Comment:
This limitation applies to access requests too, doesn't it?

Reply:
I don't know the exact use case that was desired, but the customer might have wanted the access request flow for just-in-time short lived access, vs the long term access granted from access lists.

With access requests, a user can request access, then once the request is approved they can assume the requested role through the WebUI without having to logout/login again.

To replicate this JIT flow with access lists, a user could add themselves to an access list with the desired role, but they would then need to logout/login again to obtain their new roles.